src/cardano_fmt_plug.c

```diff
@@ -0,0 +1,244 @@
+/*
+ * JtR format to crack password protected Cardano 128-byte length legacy secret Keys (a.k.a XPrv).
+ *
+ * This software is Copyright (c) 2022, Pal Dorogi <pal dot dorogi at gmail.com>
+ * and it is hereby released to the general public under the following terms:
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted.
+ *
+ */
+
+#if FMT_EXTERNS_H
+extern struct fmt_main fmt_cardano;
+#elif FMT_REGISTERS_H
+john_register_one(&fmt_cardano);
+#else
+
+#include <string.h>
+
+#ifdef _OPENMP
+#include <omp.h>
+#endif
+
+#define OMP_SCALE 1
+
+#include "arch.h"
+#include "misc.h"
+#include "common.h"
+#include "formats.h"
+#include "params.h"
+#include "options.h"
+#include "jumbo.h"
+#include "cardano_common.h"
+#include "blake2.h"
+#include "pbkdf2_hmac_sha512.h"
+#include "chacha.h"
+#include "ed25519.h"
```

Comment:
A few of these `#include` are probably irrelevant, I guess at least `#include "options.h"`.

Reply:
I have removed all unnecessary ones.

Reply:
I think we differ in what we consider "unnecessary". I meant you'd remove includes of stuff that you don't use. However, you seem to have removed everything without which the program won't build. That approach results in it building "by luck" - due to includes of other needed headers from some of the headers that you still include her. I think that's not ideal.